# Housing Market

### Introduction:

This time we will create our own dataset with fictional numbers to describe a house market. As we are going to create random data don't try to reason of the numbers.

### Step 1. Import the necessary libraries

In [1]:
import pandas as pd
import numpy as np

### Step 2. Create 3 differents Series, each of length 100, as follows: 
1. The first a random number from 1 to 4 
2. The second a random number from 1 to 3
3. The third a random number from 10,000 to 30,000

In [2]:
s1 = pd.Series(np.random.randint(1, 5, size=100))
s2 = pd.Series(np.random.randint(1, 4, size=100))
s3 = pd.Series(np.random.randint(10000, 30001, size=100))
print(s1, s2, s3)

0     1
1     2
2     2
3     4
4     2
     ..
95    2
96    4
97    3
98    1
99    4
Length: 100, dtype: int32 0     1
1     1
2     1
3     1
4     2
     ..
95    2
96    2
97    1
98    3
99    2
Length: 100, dtype: int32 0     21516
1     25320
2     14339
3     23395
4     11199
      ...  
95    14431
96    11877
97    28269
98    12163
99    14203
Length: 100, dtype: int32


### Step 3. Let's create a DataFrame by joinning the Series by column

In [3]:
df = pd.concat([s1, s2, s3], axis=1)
df.head()

,0,1,2
0,1,1,21516
1,2,1,25320
2,2,1,14339
3,4,1,23395
4,2,2,11199


### Step 4. Change the name of the columns to bedrs, bathrs, price_sqr_meter

In [4]:
df.columns = ["bedrs", "bathrs", "price_sqr_meter"]
df

,bedrs,bathrs,price_sqr_meter
0,1,1,21516
1,2,1,25320
2,2,1,14339
3,4,1,23395
4,2,2,11199
...,...,...,...
95,2,2,14431
96,4,2,11877
97,3,1,28269
98,1,3,12163


In [5]:
df.head()

,bedrs,bathrs,price_sqr_meter
0,1,1,21516
1,2,1,25320
2,2,1,14339
3,4,1,23395
4,2,2,11199


### Step 5. Create a column DataFrame with the values of the 3 Series and assign it to 'bigcolumn'

In [6]:
bigcolumn = pd.concat([s1, s2, s3], ignore_index=False)
print(bigcolumn)

0         1
1         2
2         2
3         4
4         2
      ...  
95    14431
96    11877
97    28269
98    12163
99    14203
Length: 300, dtype: int32


In [7]:
bigcolumn = pd.concat([s1, s2, s3], ignore_index=True)
print(bigcolumn)

0          1
1          2
2          2
3          4
4          2
       ...  
295    14431
296    11877
297    28269
298    12163
299    14203
Length: 300, dtype: int32


In [8]:
bigcolumn = pd.DataFrame(pd.concat([s1, s2, s3]))
print(type(bigcolumn))
bigcolumn

<class 'pandas.DataFrame'>


,0
0,1
1,2
2,2
3,4
4,2
...,...
95,14431
96,11877
97,28269
98,12163


In [9]:
bigcolumn = pd.DataFrame(pd.concat([s1, s2, s3], ignore_index=True))
print(type(bigcolumn))
bigcolumn

<class 'pandas.DataFrame'>


,0
0,1
1,2
2,2
3,4
4,2
...,...
295,14431
296,11877
297,28269
298,12163


### Step 6. Oops, it seems it is going only until index 99. Is it true?

In [10]:
bigcolumn.index.values

array([  0,   1,   2,   3,   4,   5,   6,   7,   8,   9,  10,  11,  12,
        13,  14,  15,  16,  17,  18,  19,  20,  21,  22,  23,  24,  25,
        26,  27,  28,  29,  30,  31,  32,  33,  34,  35,  36,  37,  38,
        39,  40,  41,  42,  43,  44,  45,  46,  47,  48,  49,  50,  51,
        52,  53,  54,  55,  56,  57,  58,  59,  60,  61,  62,  63,  64,
        65,  66,  67,  68,  69,  70,  71,  72,  73,  74,  75,  76,  77,
        78,  79,  80,  81,  82,  83,  84,  85,  86,  87,  88,  89,  90,
        91,  92,  93,  94,  95,  96,  97,  98,  99, 100, 101, 102, 103,
       104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116,
       117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129,
       130, 131, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142,
       143, 144, 145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 155,
       156, 157, 158, 159, 160, 161, 162, 163, 164, 165, 166, 167, 168,
       169, 170, 171, 172, 173, 174, 175, 176, 177, 178, 179, 18

In [11]:
len(bigcolumn)

300

### Step 7. Reindex the DataFrame so it goes from 0 to 299

In [12]:
help(bigcolumn.reset_index)

Help on method reset_index in module pandas.core.frame:

reset_index(
    level: 'IndexLabel | None' = None,
    *,
    drop: 'bool' = False,
    inplace: 'bool' = False,
    col_level: 'Hashable' = 0,
    col_fill: 'Hashable' = '',
    allow_duplicates: 'bool | lib.NoDefault' = <no_default>,
    names: 'Hashable | Sequence[Hashable] | None' = None
) -> 'DataFrame | None' method of pandas.DataFrame instance
    Reset the index, or a level of it.

    Reset the index of the DataFrame, and use the default one instead.
    If the DataFrame has a MultiIndex, this method can remove one or more
    levels.

    Parameters
    ----------
    level : int, str, tuple, or list, default None
        Only remove the given levels from the index. Removes all levels by
        default.
    drop : bool, default False
        Do not try to insert index into dataframe columns. This resets
        the index to the default integer index.
    inplace : bool, default False
        Whether to modify the Data

In [13]:
bigcolumn = bigcolumn.reset_index(drop=True)

In [14]:
bigcolumn

,0
0,1
1,2
2,2
3,4
4,2
...,...
295,14431
296,11877
297,28269
298,12163


In [15]:
print(type(bigcolumn[0]))
bigcolumn[0]

<class 'pandas.Series'>


0          1
1          2
2          2
3          4
4          2
       ...  
295    14431
296    11877
297    28269
298    12163
299    14203
Name: 0, Length: 300, dtype: int32

In [16]:
bigcolumn.index.values

array([  0,   1,   2,   3,   4,   5,   6,   7,   8,   9,  10,  11,  12,
        13,  14,  15,  16,  17,  18,  19,  20,  21,  22,  23,  24,  25,
        26,  27,  28,  29,  30,  31,  32,  33,  34,  35,  36,  37,  38,
        39,  40,  41,  42,  43,  44,  45,  46,  47,  48,  49,  50,  51,
        52,  53,  54,  55,  56,  57,  58,  59,  60,  61,  62,  63,  64,
        65,  66,  67,  68,  69,  70,  71,  72,  73,  74,  75,  76,  77,
        78,  79,  80,  81,  82,  83,  84,  85,  86,  87,  88,  89,  90,
        91,  92,  93,  94,  95,  96,  97,  98,  99, 100, 101, 102, 103,
       104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116,
       117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129,
       130, 131, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142,
       143, 144, 145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 155,
       156, 157, 158, 159, 160, 161, 162, 163, 164, 165, 166, 167, 168,
       169, 170, 171, 172, 173, 174, 175, 176, 177, 178, 179, 18